### DISCUSSION TITLE
How to raise a JSON:API exception with a custom status code?

### DISCUSSION THREAD

**Discussion by Nekidev** [django-json-api/django-rest-framework-json-api#1128] [Q&A] [answered]
What the title says. For example, I can raise a `ValidationError` with `forbidden` as the code, but the status code will be 400 (not 403).

```py
from rest_framework_json_api import serializers

def my_error_view(request):
    """
    This view raises an exception which will be formatted as a JSON:API error object.
    """

    raise serializers.ValidationError(
        detail="You need the `access_this_view` permission to access this resource.",
        code="forbidden"
    )
```

Any request to that endpoint in this example will return a 400 error response with a `forbidden` code. Is there any way to specify a status code?

**Comment by sliverc** [comment, marked as answer]
Django REST framework has different type of [exceptions](https://www.django-rest-framework.org/api-guide/exceptions/) for different status codes. `ValidationError` will always return a status code 400. In case you want to have 403 status code you need to raise a [PermissionDenied](https://www.django-rest-framework.org/api-guide/exceptions/#permissiondenied) exception.
